### ISSUE TITLE
App shows only a white screen when trying to see Options

### ISSUE THREAD

**Issue by ian-hamlin** [desktop/desktop#3405]
### Description

When I go to File > Options a modal is briefly displayed before the whole screen turns white.

### Version

1.0.9 (Windows 10.0.15063)

**GitHub Desktop version:** 1.0.9

Windows 10.0.15063

### Steps to Reproduce

1. Open GitHub desktop
2. Click File
3. Click Options
4. White screen

**Expected behavior:** To see the Options dialog

**Actual behavior:** The options dialog appears for a fraction of a second before the whole app screen goes white

**Reproduces how often:** Every time, I've never been able to see the options screen.

### Logs

The logs only show an app launch message.

**Comment by j-f1** [comment]
Can you open the developer tools (<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd>) and post any errors from there?

**Comment by ian-hamlin** [comment]
No errors there as far as I can see.

**Comment by shiftkey** [comment]
@ian-hamlin a couple of additional things:

> before the whole screen turns white.

I gather this is just the Desktop window and not the whole display?

> Every time, I've never been able to see the options screen.

This is the dialog where we poke at the registry to see what tools you have installed. Maaaaaybe it's related to this, but it's a long shot (and should be handling errors correctly). Is there anything interesting about your registry setup here that might help with investigating on my end? 

> No errors there as far as I can see.

Ensure you are looking at the **Console** tab and not something like **Elements** or **Sources**.

**Comment by ian-hamlin** [comment]
@shiftkey 

> I gather this is just the Desktop window and not the whole display?

Yes, sorry I was not clear, but it's just the window containg the GitHub app that goes white.

> Ensure you are looking at the Console tab and not something like Elements or Sources.

Nothing there at all, apart from a few long entries, the last entry is:

> launching: 1.0.9 (Windows 10.0.15063) install.ts:31

One thing I did notice this morning, is that a new file has appeared in the repository I'm working in with the following text (sh.exe.stackdump):

```
Exception: STATUS_STACK_OVERFLOW at rip=7FF87CB59F97
rax=0000000000001250 rbx=00000000FFFFAD90 rcx=0000000000000000
rdx=0000000180010018 rsi=0000000180271780 rdi=000000018021C490
r8 =00000000001037BE r9 =0000000180271780 r10=00000000FFFF9000
r11=00000000FFE03130 r12=00000000FFFFAC70 r13=000000018021C490
r14=00000000FFFFAC10 r15=00000000FFFFAD98
rbp=0000000000000420 rsp=00000000FFFFAA78
program=C:\Users\ian.hamlin\AppData\Local\GitHubDesktop\app-1.0.9\resources\app\git\usr\bin\sh.exe, pid 4512, thread unknown (0x6D0)
cs=0033 ds=002B es=002B fs=0053 gs=002B ss=002B
```

**Comment by shiftkey** [comment]
@ian-hamlin that file suggests something with how we're invoking Git is causing a stack overflow. If you remove that file, does it return?

**Comment by ian-hamlin** [comment]
@shiftkey The file does not return.

**Comment by ian-hamlin** [comment]
I cloned another repository and the file did come back, so it may have been a red herring.
```
Exception: STATUS_STACK_OVERFLOW at rip=7FFD895B9F97
rax=0000000000001250 rbx=00000000FFFFAD90 rcx=0000000000000000
rdx=0000000180010018 rsi=0000000180271780 rdi=000000018021C490
r8 =00000000006E38AE r9 =0000000180271780 r10=00000000FFFF9000
r11=00000000FFE03130 r12=00000000FFFFAC70 r13=000000018021C490
r14=00000000FFFFAC10 r15=00000000FFFFAD98
rbp=0000000000000420 rsp=00000000FFFFAA78
program=C:\Users\ian.hamlin\AppData\Local\GitHubDesktop\app-1.0.9\resources\app\git\usr\bin\sh.exe, pid 15912, thread unknown (0x68D0)
cs=0033 ds=002B es=002B fs=0053 gs=002B ss=002B
```

**Comment by niik** [comment]
> Nothing there at all, apart from a few long entries, the last entry is:

@ian-hamlin Could you double check to make sure that the logging level in developer tools is set to verbose?

<img width="427" alt="screen shot 2017-11-28 at 09 08 37" src="https://user-images.githubusercontent.com/634063/33308744-fbd20aee-d41b-11e7-934f-4019510121a9.png">

**Comment by ian-hamlin** [comment]
@niik It was not, after turning that on, I get the following (sorry for bad formatting)

> 10:38:49.980 renderer.js:12 [AppStore] loading 2 repositories from store
10:38:49.980 renderer.js:12 [AppStore] found account: ian-hamlin (Ian Hamlin)
10:38:49.992 renderer.js:12 Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z
10:38:50.608 renderer.js:12 Executing getBranches: git for-each-ref --format=%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(parent)%00%(symref)%00%(subject)%00%(body)%00%1F refs/heads refs/remotes
10:38:50.614 renderer.js:12 Executing getRecentBranches: git log -g --no-abbrev-commit --pretty=oneline HEAD -n 2500 --
10:38:50.807 renderer.js:5 App ready time: 20.64239400625229ms
10:38:50.808 renderer.js:5 Load time: 689.0826900005341ms
10:38:50.808 renderer.js:5 Renderer ready time: 844.6899999999999ms
10:38:50.808 renderer.js:12 Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z
10:38:50.811 renderer.js:12 Executing getBranches: git for-each-ref --format=%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(parent)%00%(symref)%00%(subject)%00%(body)%00%1F refs/heads refs/remotes
10:38:50.813 renderer.js:12 Executing getRecentBranches: git log -g --no-abbrev-commit --pretty=oneline HEAD -n 2500 --
10:38:50.851 renderer.js:12 Executing installGlobalLFSFilter: git lfs install --skip-repo
10:38:51.452 renderer.js:12 launching: 1.0.9 (Windows 10.0.15063)
10:38:51.482 renderer.js:12 Executing getCommits: git log 0983d761a2ea3633a2df543cf24874544a65a697 --date=raw --max-count=1 --pretty=%H%x1F%s%x1F%b%x1F%an <%ae> %ad%x1F%P -z --no-color --
10:38:51.488 renderer.js:12 Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z (took 1.496s)
10:38:51.488 renderer.js:12 Executing getCommits: git log 0983d761a2ea3633a2df543cf24874544a65a697 --date=raw --max-count=1 --pretty=%H%x1F%s%x1F%b%x1F%an <%ae> %ad%x1F%P -z --no-color --
10:38:51.580 renderer.js:12 Executing getWorkingDirectoryDiff: git diff --no-ext-diff --no-index --patch-with-raw -z --no-color -- /dev/null sh.exe.stackdump
10:38:51.584 renderer.js:12 Executing getCommits: git log origin/master..master --date=raw --max-count=100 --pretty=%H%x1F%s%x1F%b%x1F%an <%ae> %ad%x1F%P -z --no-color --
10:38:51.587 renderer.js:12 Executing getRemotes: git remote -v
10:38:51.591 renderer.js:12 Executing getAuthorIdentity: git var GIT_AUTHOR_IDENT
10:38:51.630 renderer.js:12 Executing getWorkingDirectoryDiff: git diff --no-ext-diff --no-index --patch-with-raw -z --no-color -- /dev/null sh.exe.stackdump
10:38:51.633 renderer.js:12 Executing getCommits: git log origin/master..master --date=raw --max-count=100 --pretty=%H%x1F%s%x1F%b%x1F%an <%ae> %ad%x1F%P -z --no-color --
10:38:51.637 renderer.js:12 Executing getRemotes: git remote -v
10:38:51.640 renderer.js:12 Executing getAuthorIdentity: git var GIT_AUTHOR_IDENT
10:38:51.835 renderer.js:12 Executing getRemotes: git remote -v
10:38:52.404 renderer.js:12 Executing installGlobalLFSFilter: git lfs install --skip-repo (took 1.553s)
10:38:55.335 renderer.js:12 Executing fetch: git -c credential.helper= fetch --progress --prune origin
10:38:56.639 renderer.js:12 Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 1.303s)
10:38:56.639 renderer.js:12 Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z
10:38:56.641 renderer.js:12 Executing getBranches: git for-each-ref --format=%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(parent)%00%(symref)%00%(subject)%00%(body)%00%1F refs/heads refs/remotes
10:38:56.643 renderer.js:12 Executing getRecentBranches: git log -g --no-abbrev-commit --pretty=oneline HEAD -n 2500 --
10:38:56.726 renderer.js:12 Executing getWorkingDirectoryDiff: git diff --no-ext-diff --no-index --patch-with-raw -z --no-color -- /dev/null sh.exe.stackdump
10:38:56.728 renderer.js:12 Executing getCommits: git log origin/master..master --date=raw --max-count=100 --pretty=%H%x1F%s%x1F%b%x1F%an <%ae> %ad%x1F%P -z --no-color --
10:38:56.731 renderer.js:12 Executing getRemotes: git remote -v
10:38:56.733 renderer.js:12 Executing getAuthorIdentity: git var GIT_AUTHOR_IDENT
10:39:02.612 renderer.js:12 Executing getIndexChanges: git diff-index --cached --name-status --no-renames -z HEAD --
10:39:02.678 renderer.js:12 Executing checkoutIndex: git checkout-index -f -u -q --stdin -z
10:39:02.751 renderer.js:12 Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z
10:39:02.753 renderer.js:12 Executing getBranches: git for-each-ref --format=%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(parent)%00%(symref)%00%(subject)%00%(body)%00%1F refs/heads refs/remotes
10:39:02.755 renderer.js:12 Executing getRecentBranches: git log -g --no-abbrev-commit --pretty=oneline HEAD -n 2500 --
10:39:02.832 renderer.js:12 Executing getCommits: git log origin/master..master --date=raw --max-count=100 --pretty=%H%x1F%s%x1F%b%x1F%an <%ae> %ad%x1F%P -z --no-color --
10:39:02.836 renderer.js:12 Executing getRemotes: git remote -v
10:39:02.839 renderer.js:12 Executing getAuthorIdentity: git var GIT_AUTHOR_IDENT
10:39:18.378 install.ts:35 Executing getConfigValueInPath: git config -z --global user.name
10:39:18.461 install.ts:35 Executing getConfigValueInPath: git config -z --global user.email
10:39:18.540 install.ts:35 Executing getConfigValueInPath: git config -z --global merge.tool

**Comment by niik** [comment]
Alright, so it looks to me like the Git commands are running just fine and nothing else looks out of place as far as I can see.

@ian-hamlin Do other dialogs render okay? Like, can you open repository settings (under the repository menu)?

**Comment by ian-hamlin** [comment]
@niik Yes, that dialog opens just fine.  

I've noticed that sometimes it takes a little longer to crash, at first I thought it was immediately, but today I was able to see the dialog for a good few seconds before it all went white.

**Comment by shiftkey** [comment]
While I'm working on porting our registry access code so that it works under restrictive group policy rules I was able to reproduce this behaviour - it feels like a crash but there's no clues in the dev tools or main process as to why it's crashing.

I'll assign myself to this to dig into surfacing better errors while working on this, and hopefully the fix for #3105 also fixes this for you.
